addons/mail_bot/models/res_users.py

```diff
@@ -25,13 +26,16 @@ class ResUsers(models.Model):
     def SELF_READABLE_FIELDS(self):
         return super().SELF_READABLE_FIELDS + ['odoobot_state']
 
-    def _init_messaging(self, store):
-        odoobot_onboarding = False
-        if self.odoobot_state in [False, 'not_initialized'] and self._is_internal():
-            odoobot_onboarding = True
+    def _on_webclient_bootstrap(self):
+        super()._on_webclient_bootstrap()
+        if self._is_internal() and not self.odoobot_state or self.odoobot_state == "not_initialized":
             self._init_odoobot()
-        super()._init_messaging(store)
-        store.add({"odoobotOnboarding": odoobot_onboarding})
+
+    def _init_store_data(self, store: Store, /):
+        store.add({"odoobotOnboarding": False})
+        if self._is_internal() and not self.odoobot_state or self.odoobot_state == "not_initialized":
+            store.add({"odoobotOnboarding": True})
```

Comment:
This condition is actually no longer correct. It used to be called before `_init_odoobot` so assumed it set True when it became initialized, but this way it is the inverse of what we want.

`odoobotOnboarding` should be True in any of the intermediate step, so exclude `False`, `not_initialized`, `disabled`, and maybe `idle` (I have to double check that one).

I'll push the last fixes so we can more forward.